src/libYARP_dev/src/MapGrid2D.cpp

```diff
@@ -342,7 +342,7 @@ bool MapGrid2D::loadMapYarpAndRos(string yarp_filename, string ros_yaml_filename
             for (size_t x = 0; x < m_width; x++)
             {
                 yarp::sig::PixelRgb pix_occ = ros_img.safePixel(x, y);
-                double color_avg = (pix_occ.r + pix_occ.g + pix_occ.b) / 3;
+                int color_avg = (pix_occ.r + pix_occ.g + pix_occ.b) / 3;
```

Comment:
perhaps instead of changing this to 'int', change the '3' to '3.0'?

Reply:
My two cents here.
This is a mean on rgb color, where the three components are `unsigned int8`, and imho the result shall  be `unsigned int8` as well, since the output still is a color. It will be converted anyway in the following file. I don't think double precision calculations is a requirement here, so I'd choose the fastest way since it has to be done for all pixels in the image.

Also I'd move the declaration of `color_avg` and `occ` before the `for`

Reply:
actually its name is `color_avg`, therefore an average, not a color... 🙄 
Anyway the speed requirement seems like a good reason for changing it to unsigned int8

Reply:
I have a question about this piece of code: in the following line there is a double 0≤ n ≤1 which is going to be casted to unsigned char. I may be wrong so please correct me but it seems to me that `occ` will always be 0 unless `color_avg` is 0, so that `occ` would be 1. Is this the expected behaviour?

Reply:
@damn1 I think you are right. Probably the logic is to draw on the image the presence/absence of an obstacle and uses grayscale as a visual _bool_ representation, but in this case a simple test like `color_avg != 0` would be more clear. I think we shall ask the writer of this code what was the purpose.

Reply:
CC @randaz81